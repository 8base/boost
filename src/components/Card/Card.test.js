// @flow
import React from 'react';
import { Card } from './';

describe('<Card />', () => {
  it('should render Card components', () => {
    const wrapper = mount(
      <Card padding="sm">
        <Card.Header padding="md">Header</Card.Header>
        <Card.Body padding="lg">
          <Card.Section>
            Body
          </Card.Section>
        </Card.Body>
        <Card.Footer padding="xl">Footer</Card.Footer>
      </Card>,
    );

    expect(wrapper.find(Card.Header).text()).toBe('Header');
    expect(wrapper.find(Card.Body).text()).toBe('Body');
    expect(wrapper.find(Card.Footer).text()).toBe('Footer');

    expect(wrapper.find(Card.Header).props().padding).toBe('md');
    expect(wrapper.find(Card.Body).props().padding).toBe('lg');
    expect(wrapper.find(Card.Footer).props().padding).toBe('xl');

    expect(wrapper).toMatchSnapshot();
  });


  it('should render Card components with render prop', () => {
    const wrapper = mount(
      <BoostProvider>
        <Card someArg={ 42 } padding="xl">
          { ({ someArg }: *) => (
            <React.Fragment>
              <Card.Header>Header</Card.Header>
              <Card.Body scrollable>
                Body { someArg }
              </Card.Body>
              <Card.Footer>Footer</Card.Footer>
            </React.Fragment>
          ) }
        </Card>
      </BoostProvider>,
    );

    expect(wrapper.find(Card.Body).text()).toBe('Body 42');
  });
});

