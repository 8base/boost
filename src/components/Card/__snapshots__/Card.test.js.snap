// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render Card components 1`] = `
.emotion-16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-height: 100%;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 64px;
}

.emotion-12 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

<Card
  padding="sm"
>
  <Paper>
    <Styled(paper)
      tagName="div"
    >
      <Tag
        className="emotion-16 emotion-1"
        tagName="div"
      >
        <div
          className="emotion-16 emotion-1"
        >
          <CardHeader
            key=".0"
            padding="sm"
          >
            <Styled(cardHeader)
              padding="sm"
              tagName="div"
            >
              <Tag
                className="emotion-0 emotion-1"
                padding="sm"
                tagName="div"
              >
                <div
                  className="emotion-0 emotion-1"
                >
                  Header
                </div>
              </Tag>
            </Styled(cardHeader)>
          </CardHeader>
          <CardBody
            key=".1"
            padding="sm"
          >
            <Styled(cardBody)
              padding="sm"
              tagName="div"
            >
              <Tag
                className="emotion-8 emotion-1"
                padding="sm"
                tagName="div"
              >
                <div
                  className="emotion-8 emotion-1"
                >
                  <CardSection
                    padding="md"
                  >
                    <Styled(cardSection)
                      padding="md"
                      tagName="div"
                    >
                      <Tag
                        className="emotion-4 emotion-1"
                        padding="md"
                        tagName="div"
                      >
                        <div
                          className="emotion-4 emotion-1"
                        >
                          Body
                        </div>
                      </Tag>
                    </Styled(cardSection)>
                  </CardSection>
                </div>
              </Tag>
            </Styled(cardBody)>
          </CardBody>
          <CardFooter
            key=".2"
            padding="sm"
          >
            <Styled(cardFooter)
              padding="sm"
              tagName="div"
            >
              <Tag
                className="emotion-12 emotion-1"
                padding="sm"
                tagName="div"
              >
                <div
                  className="emotion-12 emotion-1"
                >
                  Footer
                </div>
              </Tag>
            </Styled(cardFooter)>
          </CardFooter>
        </div>
      </Tag>
    </Styled(paper)>
  </Paper>
</Card>
`;
